authenticate-oauth/Web/Authenticate/OAuth.hs

```diff
@@ -307,16 +319,24 @@ getAccessTokenProxy, getTokenCredentialProxy
                -> m Credential -- ^ Token Credential (Access Token & Secret)
 getAccessTokenProxy p = getAccessToken' $ addMaybeProxy p
 
-
 getAccessToken' :: MonadIO m
                 => (Request -> Request)       -- ^ Request Hook
                 -> OAuth                      -- ^ OAuth Application
                 -> Credential                 -- ^ Temporary Credential (with oauth_verifier if >= 1.0a)
                 -> Manager
                 -> m Credential     -- ^ Token Credential (Access Token & Secret)
-getAccessToken' hook oa cr manager = do
+getAccessToken' = getAccessToken'' addAuthHeader
+
+getAccessToken'' :: MonadIO m
```

Comment:
I'm open to suggestions (or just changing the signature on getAccessToken' if thats permissible). This is the most flexible version of getAccessToken is all.